test/worker_test.rb

```diff
@@ -720,6 +720,23 @@ def self.perform
     assert_equal 1, Resque.workers.size
   end
 
+  it "prunes workers that haven't been registered but have set a heartbeat" do
+    assert_equal({}, Resque::Worker.all_heartbeats)
+    now = Time.now
+
+    workerA = Resque::Worker.new(:jobs)
+    workerA.to_s = "bar:3:jobs"
+    workerA.heartbeat!(now - Resque.prune_interval - 1)
+
+    assert_equal 0, Resque.workers.size
+    assert Resque::Worker.all_heartbeats.key?(workerA.to_s)
```

Comment:
Would we also want to test that `Worker.all` is empty?

Reply:
Added to L733